src/RequestSender.ts

```diff
@@ -555,7 +567,12 @@ export class RequestSender {
         prepareAndMakeRequest
       );
     } else {
-      prepareAndMakeRequest(null, stringifyRequestData(data || {}));
+      prepareAndMakeRequest(
+        null,
+        options.apiMode === 'preview'
+          ? JSON.stringify(data || {}, dateTimeReplacer)
```

Comment:
Let's extract this + dateTimeReplacer into a separate function. Feels a bit untidy to have `dateTimeReplacer` floating around.

Reply:
I extracted this to `jsonStringifyRequestData` defined in utils.ts, and I also renamed the existing `stringifyRequestData` -> `queryStringifyRequestData` to better differentiate between the two.